components/Icon.js

```diff
@@ -0,0 +1,24 @@
+import styled from 'styled-components';
+
+const icons = {
+    close: '\\e903',
+    hamburger: '\\e90b',
+};
+
+const Icon = styled.span`
+    font-family: 'icomoon' !important;
+    speak: none;
+    font-style: normal;
+    font-weight: normal;
+    font-variant: normal;
+    text-transform: none;
+    line-height: 1;
+    -webkit-font-smoothing: antialiased;
+    -moz-osx-font-smoothing: grayscale;
```

Comment:
The not so cool thing about styled component is that the above definitions are duplicating for every single variant of this component, although they are always the same... I have also tried to do something like the following, but it didn't really help:

```javascript
const IconContainer = styled.span`
    font-family: 'icomoon' !important;
    ...
`;

const Icon = styled(IconContainer)`
    &:before {
         content: "${(props) => icons[props.name]}";
     }
`;
```

@tomsvogel Any idea how to solve that issue?